examples/basic/src/util/createKurt.ts

```diff
@@ -0,0 +1,29 @@
+import { Kurt, type KurtAdapter } from "@formula-monks/kurt"
+
+import { KurtOpenAI } from "@formula-monks/kurt-open-ai"
+import { OpenAI } from "openai"
+
+import { KurtVertexAI } from "@formula-monks/kurt-vertex-ai"
+import { VertexAI } from "@google-cloud/vertexai"
+
+const DEFAULT_MODEL = "gpt-4o"
+
+export const createKurt = (model: string | undefined): Kurt => {
+  const adapter = findAdapter(model ?? DEFAULT_MODEL)
+  if (!adapter) throw new Error(`Model ${model} is not supported.`)
+  return new Kurt(adapter)
+}
+
+const findAdapter = (model: string): KurtAdapter | undefined => {
+  if (KurtOpenAI.isSupportedModel(model))
+    return new KurtOpenAI({ openAI: new OpenAI(), model })
+
+  if (KurtVertexAI.isSupportedModel(model))
+    return new KurtVertexAI({
+      vertexAI: new VertexAI({
+        project: process.env.VERTEX_AI_PROJECT ?? "my-project",
+        location: process.env.VERTEX_AI_LOCATION ?? "us-central1",
+      }),
+      model,
+    })
```

Comment:
I'll leave these decisions to you, but I just wanted to mention that I personally prefer returning `null` explicitly from functions like these, because `undefined` is what is return by default from `void` functions, early return statements and so on, is less explicit and can mask issues.

Reply:
I'm very much in the "null is bad" camp and I avoid it wherever I can.

That is, I think that `null` and `undefined` are very closely overlapping ideas - close enough that there is often confusion about which to use, but different enough in that they are not interchangeable . This leads to situatiions where people need to make rather subtle tradeoff decisions, often leading to subtle bugs if somebody was expecting one but got the other. Because they are so similar, yet different enough to cause trouble, I (and many others in the JS/TS community) think it's beneficial to basically "ban" one of them from our code patterns, as much as possible.

If people want to ban one of them, the one people usually ban is `null`. In fact, even [Microsoft's style guide for contributing to the TypeScript project does](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined).

The reasons for this are:
- `undefined` is super difficult to get rid of, being baked in to the fundamental concepts of optional/missing values
- relatedly, `undefined` is usually nicely ergonomic in TypeScript because of how it is baked in to optional/missing values
- `typeof null === 'object'` is fundamentally stupid baggage to be saddled with

I can see your point about wanting to be explicit here about having a `return null`, but ultimately I think I land on the side that an implicit `return undefined` is good, actually - if you didn't return anything, then it falls back to `undefined`. I think it's nice that way, but I can see how one would prefer the explicit way. Though even if I preferred the explicit way here, I think it would get overridden by my stronger desire to "ban" `null` from my code for the reasons mentioned above.

Reply:
Thanks for explaining your point of view! I agree with the reasoning there, and respect the decision, I intended to create a PR adding a rule to forbid nulls, but Biome doesn't seem to support adding new, custom rules, I guess we need to rely on code reviews for this one.